meson.build

```diff
@@ -429,6 +429,7 @@ if get_option('buildtype') != 'debug'
         possible_cc_flags += [
                 '-ffunction-sections',
                 '-fdata-sections',
+                '-fno-semantic-interposition',
```

Comment:
I think we should do this unconditionally, i.e. not exclude debug mode from this. With the exclusion, the code we test in debug mode could behave significantly differently from the release code, which we don't want.

Reply:
i agree, this should just be done always

Reply:
ok, let me check.

Reply:
did as requested